prov/sockets/src/sock_av.c

```diff
@@ -578,8 +578,8 @@ int sock_av_open(struct fid_domain *domain, struct fi_av_attr *attr,
 
 		if (_av->table_hdr == MAP_FAILED) {
 			SOCK_LOG_ERROR("mmap failed\n");
-			free(_av);
 			shm_unlink(_av->name);
+			free(_av);
```

Comment:
@jithinjosepkl - These changes look correct.  Jithin, there should be a single exit location where error handling occurs, so that the cleanup code is not duplicated.  This should be true throughout the code base.